File: 1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
        String[] strArr = s.split(" ");
        String[] sortedStr = new String[strArr.length];

        for(String word : strArr) {
            char [] ch = word.toCharArray();
            int i = Integer.parseInt(word.substring(ch.length - 1));
            sortedStr[i - 1] = word.substring(0, ch.length - 1);
        }
        
        return String.join(" ", sortedStr);
    }
}

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.js
/**
 * @param {string} s
 * @return {string}
 */
var sortSentence = function(s) {
    
    
    return s.split(" ").sort((a,b)=>a[a.length-1]-b[b.length-1]).map((v)=>{ 
        return v.slice(0,v.length-1)
    }).join(" ")
};